playbooks/cnf/deploy-run-cnf-tests-script.yaml

```diff
@@ -0,0 +1,191 @@
+# ginkgo_additional_params          # string: Additional Ginkgo parameters
+# features                          # string: List of features to test, separated by '|'
+# role_worker_cnf                   # string: CNF role label
+# cnf_test_dir                      # string: Path to the cnf-feature-deploy project
+# sctp_host_non_cnf_workers         # bool: Whether SCTP is enabled for non-CNF worker nodes
+# oo_install_ns                     # string: Namespace where MetalLB is deployed
+# cnf_interfaces                    # string: List of CNF SR-IOV interfaces to be tested, separated by ','
+# cnf_test_perf_test_profile        # string: Name of the pre-installed performance profile
+
+- name: Set up cnf-tests project
+  hosts: bastion
+  gather_facts: true
+  vars:
+    cnf_test_repo: "https://github.com/openshift-kni/cnf-features-deploy.git"
+    cnf_test_dir: /tmp/
+    cnftests_git_dest: cnf-features-deploy
+    container_registry: "{{ internal_container_registry_name }}/rh-osbs"
+    features: "sctp"
+    role_worker_cnf: "workercnf"
+  tasks:
+
+    - name: Gather cluster network information
+      register: cluster_info
+      environment:
+        K8S_AUTH_KUBECONFIG: "{{ kubeconfig }}"
+      kubernetes.core.k8s_info:
+        api: operator.openshift.io/v1
+        kind: Network
+        name: cluster
+
+    - name: Extract OpenShift version details
+      ansible.builtin.set_fact:
+        version: "{{ cluster_info.resources[0].status.version.split('.')[0] }}.{{ cluster_info.resources[0].status.version.split('.')[1] }}"
+
+    - name: Print cluster version
+      ansible.builtin.debug:
+        msg: "Cluster version: {{ version }}"
+
+    - name: Ensure repository does not exist
+      ansible.builtin.file:
+        path: "{{ cnf_test_dir }}/{{ cnftests_git_dest }}"
+        state: absent
+
+    - name: Clone cnf-feature-deploy project
+      block:
+        - name: Git clone cnf-feature-deploy repo specific branch
+          ansible.builtin.git:
+            repo: "{{ cnf_test_repo }}"
+            dest: "{{ cnf_test_dir }}/{{ cnftests_git_dest }}"
+            force: true
+            update: true
+            recursive: true
+            version: "release-{{ version }}"
+          register: cnftests_git_clone
+          retries: 3
+          delay: 5
+          until: cnftests_git_clone is not failed
+
+      rescue:
+        - name: Git clone cnf-feature-deploy repo master branch
+          ansible.builtin.git:
+            repo: "{{ cnf_test_repo }}"
+            dest: "{{ cnf_test_dir }}/{{ cnftests_git_dest }}"
+            force: true
+            update: true
+            recursive: true
+            version: "master"
+          register: cnftests_master_git_clone
+          retries: 3
+          delay: 5
+          until: cnftests_master_git_clone is not failed
+
+    - name: Install Ginkgo v2
+      changed_when: false
+      ansible.builtin.shell: |
+        go install github.com/onsi/ginkgo/v2/ginkgo@v2.23.3
```

Comment:
Maybe we should have ginkgo version as a variable?

Reply:
+1